app/src/main/java/com/hackadroid/pingu/com/hackadroid/pingu/activity/Intro3.java

```diff
@@ -0,0 +1,226 @@
+package com.hackadroid.pingu.com.hackadroid.pingu.activity;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Settings;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import org.apache.log4j.Logger;
+
+import com.hackadroid.pingu.R;
+
+import static android.support.design.R.attr.title;
+
+public class Intro3 extends AppCompatActivity {
+    private static final Logger log3 = Logger.getLogger(MapViewFragment.class);
+    public int GPSoff = 0;
+    /* GPS Constant Permission */
+    private static final int MY_PERMISSION_ACCESS_COARSE_LOCATION = 11;
+    private static final int MY_PERMISSION_ACCESS_FINE_LOCATION = 12;
+
+    /* Position */
+    private static final int MINIMUM_TIME = 10000;  // 10s
+    private static final int MINIMUM_DISTANCE = 50; // 50m
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_intro3);
+
+        Button b = (Button) findViewById(R.id.b1);
+        b.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                log3.info("Inside Button click !");
+                Toast.makeText(getApplicationContext(), "button activity", Toast.LENGTH_LONG).show();
+                checkConnection();
+                findloc();
+                           }
+
+        });
+
+    }
+
+    private void checkConnection() {
```

Comment:
Is this function only used for checking GPS ? If so, pls change the function name..
A function should return meaningful response. In this case, I see that this is used to check for some connection - so it should return a boolean value stating if the connection is established or not.
Break the function is multiple functions if you feel that you would use them in multiple places